src/Iris/ProofMode/Tactics/Revert.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2025 Oliver Soeser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Soeser
+-/
+import Iris.ProofMode.Tactics.Cases
+
+namespace Iris.ProofMode
+open Lean Elab Tactic Meta Qq BI Std
+
+theorem wand_revert [BI PROP] {P Q A1 A2 : PROP}
+    (h1 : P ⊣⊢ A1 ∗ A2) (h2 : A1 ⊢ A2 -∗ Q) : P ⊢ Q :=
+  h1.mp.trans (wand_elim h2)
+
+theorem forall_revert [BI PROP] {P : PROP} {Ψ : α → PROP}
+    (h : P ⊢ ∀ x, Ψ x) : ∀ x, P ⊢ Ψ x :=
+  λ x => h.trans (forall_elim x)
+
+theorem pure_revert [BI PROP] {P Q : PROP} {φ : Prop}
+    (h : P ⊢ ⌜φ⌝ -∗ Q) : φ → P ⊢ Q :=
```

Comment:
This should use the `MakeAffinely ⌜ φ ⌝ P` to add the `<affine>` modality in front of ` ⌜ φ ⌝` for logics that require it. Otherwise, `irevert` can make the goal unprovable. See 
https://gitlab.mpi-sws.org/iris/iris/-/blob/306d8d024f9d6522c0e733fd7eb27b0af676d4ca/iris/proofmode/coq_tactics.v